bundles/org.eclipse.ui.navigator/src/org/eclipse/ui/internal/navigator/actions/LinkEditorAction.java

```diff
@@ -58,61 +55,49 @@ public class LinkEditorAction extends Action implements ISelectionChangedListene
 	private boolean ignoreSelectionChanged;
 	private boolean ignoreEditorActivation;
 
-	private UIJob activateEditorJob = new UIJob(CommonNavigatorMessages.Link_With_Editor_Job_) {
-		@Override
-		public IStatus runInUIThread(IProgressMonitor monitor) {
-
-			if (!commonViewer.getControl().isDisposed()) {
-				IStructuredSelection selection = commonViewer.getStructuredSelection();
-				if (selection != null && !selection.isEmpty()) {
-
-					if (selection.size() == 1) {
-						ILinkHelper[] helpers = linkService.getLinkHelpersFor(selection.getFirstElement());
-						if (helpers.length > 0) {
-							ignoreEditorActivation = true;
-							SafeRunner.run(new NavigatorSafeRunnable() {
-								@Override
-								public void run() throws Exception {
-									helpers[0].activateEditor(commonNavigator.getSite().getPage(), selection);
-								}
-							});
-							ignoreEditorActivation = false;
+	private UIJob activateEditorJob = UIJob.create(CommonNavigatorMessages.Link_With_Editor_Job_, m -> {
+		LinkEditorAction thisAction = this;
```

Comment:
`LinkEditorAction thisAction = this;` looks like a hack to avoid compiler error if it wouldn't be a hack.

I would prefer not to refactor **this** code or do it in a way that avoids "hacking"

Reply:
That's right, without this there is a compiler-error that the fields used below are not yet initialized.
I assume that this is because the lambda is then bound to the fields instead of `this`.
I don't know a better way at the moment and reverted that refactoring for now. If somebody else knows a better way it can be applied in the future.